client/my-sites/invites/invite-accept/index.jsx

```diff
@@ -62,17 +63,21 @@ export default React.createClass( {
 		);
 	},
 
-	getRedirectTo() {
+	getRedirectAfterAccept() {
 		const { invite } = this.state.invite;
-		let redirectTo = window.location.origin;
 		switch ( invite.meta.role ) {
 			case 'viewer':
 			case 'follower':
+				return '/';
 				break;
 			default:
-				redirectTo += '/posts/' + invite.blog_id;
+				return '/posts/' + invite.blog_id;
 		}
-		return redirectTo += '?invite_accepted=' + invite.blog_id;
+	},
+
+	decline() {
+		page( '/' );
+		displayInviteDeclined();
```

Comment:
I noticed a bit of jankiness with this notice. At times the notice will appear on the invite accepted page and then show again on the the reader. I would think that we should only show the notice on the reader. I haven't looked at the implementation behind this, but I recall that we have an argument to show notices on the next page. Perhaps we should use that here?

![screen shot 1](https://cloud.githubusercontent.com/assets/1126811/11541324/55c39232-98f7-11e5-9a4f-cc3bc0e712e6.png)